### PR TITLE
♻️ (rc): Separate timeout: inner state and state transition

### PR DESCRIPTION
**Pull request by YannLocatelli**

### CODE REVIEW

File: app/os/main.cpp [hunk 1]
```diff
@@ -408,15 +408,17 @@ namespace robot {
 
 	namespace internal {
 
-		auto sleep_timeout = CoreTimeout {};
+		auto inner_state_timeout	  = CoreTimeout {};
+		auto state_transition_timeout = CoreTimeout {};
```
Comment (ladislas) [inline app/os/main.cpp:412]: small naming suggestion:

```suggestion
		auto timeout_state_internal = CoreTimeout {};
		auto timeout_state_transition  = CoreTimeout {};
```

File: libs/RobotKit/include/RobotController.h [hunk 2]
```diff
@@ -498,9 +497,11 @@ class RobotController : public interface::RobotController
   private:
 	system::robot::sm::logger logger {};
 
+	interface::Timeout &_inner_state_timeout;
+
```
Comment (ladislas) [inline libs/RobotKit/include/RobotController.h:501]:
```suggestion
	interface::Timeout &_timeout_state_internal;
	interface::Timeout &_timeout_state_transition;
```

**Review by ladislas — APPROVED** [review]
Thanks a lot! LGTM 👍 
just a small naming suggestion.

**Review by HPezz — APPROVED** [review]
LGTM

**Review by ladislas — APPROVED** [review]

**Pull request merged**